### ISSUE TITLE
关于对比学习损失函数

### ISSUE THREAD

**Issue by XiongjunGuan** [WooyoohL/MER2024-SEMI#1]
您好，我最近在学习您的代码。我发现关于对比学习部分，您在文档中设置的损失函数，按照我的理解只是计算了每个样本的特征自身和自身的相关性，并没有考虑样本之间的相互关系(即没有产生对比效果)。按照我的理解，这似乎和文章中的描述不太一致。
另外，我按照自己的理解重新实现了对比学习损失，发现 loss 完全不下降，能否帮忙解惑？非常感谢！

原本的损失函数如下：
```
def nt_xent_loss(self, z_i, z_j, temperature):
        z = torch.cat([z_i, z_j], dim=0)
        sim_matrix = torch.mm(z, z.T) / temperature
        sim_matrix /= temperature
        logits_max, _ = torch.max(sim_matrix, dim=1, keepdim=True)
        logits = sim_matrix - logits_max.detach()
        mask = torch.eye(z.size(0), dtype=torch.bool, device=z.device)
        labels = torch.arange(z.size(0), device=z.device)
        labels = torch.cat([labels[z_i.size(0):], labels[:z_j.size(0)]], dim=0)
        log_probs = F.log_softmax(logits, dim=1)
        loss = -log_probs[mask]
        return loss.mean()
```

我自己实现的版本如下：
```
def nt_xent_loss(self, z_i, z_j, temperature):
        b, c = z_i.shape

        z = torch.cat([z_i, z_j], dim=0)

        # Cosine similarity
        xcs = torch.mm(z, z.T) / temperature
        xcs[torch.eye(z.size(0)).bool()] = float("-inf")

        # Ground truth labels
        target1 = torch.arange(b, 2 * b, 1)
        target2 = torch.arange(0, b, 1)
        target = torch.cat([target1, target2], dim=0).cuda()

        # Standard cross entropy loss
        return F.cross_entropy(xcs / temperature, target, reduction="mean")
```
